src/Chapter4.hs

```diff
@@ -488,7 +497,19 @@ Implement the 'Applicative' instance for our 'List' type.
   may also need to implement a few useful helper functions for our List
   type.
 -}
+instance Semigroup (List a) where
+  (<>) :: List a -> List a -> List a
+  Empty <> xs = xs
+  Cons x xs <> ys = Cons x (xs <> ys)
```

Comment:
Wow, that's very clever! Nice idea and solution :100:

Reply:
I also tried to implement the `mconcat` using the Monoid class, instead of the concatList function I did below. The problem is that `mconcat` expected `[List a]` and not `List (List a)` :sweat_smile:

Reply:
Aha, I see :smile: `mconcat` is a very old method, so it's not general enough. If you want to `concat` the value of type `List (List a)`, you can use the `fold` function, but it requires to implement the `Foldable` instance for the list, which is a separate story...